week-3/Homework/mandatory/7-password-validator.js

```diff
@@ -21,49 +21,59 @@ Expected Result:
 PasswordValidationResult=  [false, false, false, false, true]
 
 */
+function validatePassword(pass) {
+  let regexUpper = /[A-Z]/;
+  let regexLower = /[a-z]/;
+  let regexNum = /[0-9]/;
+  let regexSymbol = /[!#$%.]/;
 
-function validatePasswords(passwords) {
+  return (
+    pass.length >= 5 &&
+    regexUpper.test(pass) &&
+    regexLower.test(pass) &&
+    regexNum.test(pass) &&
+    regexSymbol.test(pass)
+  );
+}
 
+function validatePasswords(passwords) {
+  return passwords.map(password => validatePassword(password));
 }
 
 /* ======= TESTS - DO NOT MODIFY ===== */
 
-const passwords1 = ["Se%5", "TktE.TJTU", "384#HsHF", "dvyyeyy!5", "tryT3729"]
-const passwords2 = ["StUFf27%", "Pl3nty!", "Jai33", "shajsaUA**&&", "Pl3nty!"]
+const passwords1 = ["Se%5", "TktE.TJTU", "384#HsHF", "dvyyeyy!5", "tryT3729"];
+const passwords2 = ["StUFf27%", "Pl3nty!", "Jai33", "shajsaUA**&&", "Pl3nty!"];
 
 function arraysEqual(a, b) {
-    if (a === b) return true;
-    if (a == null || b == null) return false;
-    if (a.length != b.length) return false;
-  
-    for (let i = 0; i < a.length; ++i) {
-      if (a[i] !== b[i]) return false;
-    }
-  
-    return true;
+  if (a === b) return true;
+  if (a == null || b == null) return false;
+  if (a.length != b.length) return false;
+
+  for (let i = 0; i < a.length; ++i) {
+    if (a[i] !== b[i]) return false;
+  }
+
+  return true;
 }
 
 function test(test_name, expr) {
-    let status;
-    if (expr) {
-      status = "PASSED";
-    } else {
-      status = "FAILED";
-    }
-  
-    console.log(`${test_name}: ${status}`);
+  let status;
+  if (expr) {
+    status = "PASSED";
+  } else {
+    status = "FAILED";
+  }
+
+  console.log(`${test_name}: ${status}`);
 }
 
 test(
-   "validatePasswords function works - case 1",
-   arraysEqual(
-      validatePasswords(passwords1), [false, false, true, false, false]
-   )
- );
-
- test(
-   "validatePasswords function works - case 2",
-   arraysEqual(
-      validatePasswords(passwords2), [true, true, false, false, false]
-   )
- );
+  "validatePasswords function works - case 1",
+  arraysEqual(validatePasswords(passwords1), [false, false, true, false, false])
+);
+
+test(
+  "validatePasswords function works - case 2",
+  arraysEqual(validatePasswords(passwords2), [true, true, false, false, true])
```

Comment:
I realised you shouldn't change that last `false` to `true`, notice the exercise says:

> Passwords must not be any previous password in the passwords array.

Reply:
Ohhh I see! Thanks for pointing this out @frosas I will fix it.